tests/suite/test_transport_server_tcp_load_balance.py

```diff
@@ -249,3 +251,69 @@ def test_tcp_request_load_balanced_missing_service(
             client.close()
 
         self.restore_ts(kube_apis, transport_server_setup)
+
+    def make_holding_connection(self, host, port):
+        print(f"sending tcp requests to: {host}:{port}")
+        client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client.connect((host, port))
+        client.sendall(b'hold')
+        response = client.recv(4096)
+        endpoint = response.decode()
+        print(f'response: {endpoint}')
+        return client
+
+    def test_tcp_request_max_connections(
+            self, kube_apis, crd_ingress_controller, transport_server_setup, ingress_controller_prerequisites
+    ):
+        """
+        Requests to the load balanced TCP service should result in responses from 3 different endpoints.
+        """
+
+        # step 1 - set max connections to 2 with 1 replica
+        patch_src = f"{TEST_DATA}/transport-server-tcp-load-balance/max-connections-transport-server.yaml"
+        patch_ts(
+            kube_apis.custom_objects,
+            transport_server_setup.name,
+            patch_src,
+            transport_server_setup.namespace,
+        )
+        wait_before_test()
+
+        # step 2 - make the number of allowed connections
+        port = transport_server_setup.public_endpoint.tcp_server_port
+        host = transport_server_setup.public_endpoint.public_ip
+
+        clients = []
+        for i in range(6):
```

Comment:
Is there a reason the max possible concurrent connections is 6 here but yaml says `maxConnections: 2`

Reply:
there's 6 here as there's 3 replicas of 2 max connections.